File: Jump Game - Leetcode 55/Jump Game - Leetcode 55.java
class Solution {
    public boolean canJump(int[] nums) {
        int n = nums.length;
        int target = n - 1;

        for (int i = n - 1; i >= 0; i--) {
            if (i + nums[i] >= target) {
                target = i;
            }
        }

        return target == 0;
    }
}

File: Jump Game - Leetcode 55/Jump Game - Leetcode 55.js
var canJump = function(nums) {
    const n = nums.length;
    let target = n - 1;

    for (let i = n - 1; i >= 0; i--) {
        if (i + nums[i] >= target) {
            target = i;
        }
    }

    return target === 0;
};
